File: app/src/main/java/com/msaggik/algorithmsdatastructures/java/sort/BubbleSort.java
package com.msaggik.algorithmsdatastructures.java.sort;

/**
 * Сортировка "Пузырьком"
 * <p>
 * Вычислительная сложность: О(N^2)
 * Затраты памяти: O(1)
 **/

public class BubbleSort {

    public static int[] bubbleSort(int[] arrayInt) {
        for (int i = 0; i < arrayInt.length - 1; i++)
            for (int j = i + 1; j < arrayInt.length; j++) {
                if(arrayInt[i] > arrayInt[j]) {
                    int value = arrayInt[i];
                    arrayInt[i] = arrayInt[j];
                    arrayInt[j] = value;
                }
            }
        return arrayInt;
    }
}

File: app/src/main/java/com/msaggik/algorithmsdatastructures/kotlin/sort/BubbleSort.kt
package com.msaggik.algorithms.kotlin.sort

object BubbleSort {
    fun bubbleSort(arrayInt: IntArray): IntArray {
        for (i in 0 until arrayInt.size - 1) for (j in i + 1 until arrayInt.size) {
            if (arrayInt[i] > arrayInt[j]) {
                val value = arrayInt[i]
                arrayInt[i] = arrayInt[j]
                arrayInt[j] = value
            }
        }
        return arrayInt
    }
}